src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -882,6 +882,10 @@ public function getDefinition($id)
             throw new InvalidArgumentException(sprintf('The service definition "%s" does not exist.', $id));
         }
 
+        if ($this->definitions[$id]->isDeprecated()) {
```

Comment:
I wasn't sure till now if I should trigger that error on the service instanciation or the definition getter, but then I'll do it in the service instanciation then.